.github/workflows/download-pulumi-cron.yml

```diff
@@ -0,0 +1,141 @@
+name: Download Pulumi Cron
+"on":
+  schedule:
+    - cron: 0 8 * * *
+
+jobs:
+  macos-homebrew-install:
+    name: Install Pulumi with Homebrew on macOS
+    runs-on: macos-latest
+    steps:
+      - name: homebrew install
+        run: brew install pulumi
+      - name: Pulumi Version Details
+        id: vars
+        run: |
+          echo ::set-output name=installed-version::$(pulumi version)
+          echo ::set-output name=expected-version::v$(curl -sS https://www.pulumi.com/latest-version)
+      - name: Error if incorrect version found
+        if: steps.vars.outputs.expected-version != steps.vars.outputs.installed-version
+        run: |
+          echo "Expected version ${{ steps.vars.outputs.expected-version }} but found ${{ steps.vars.outputs.installed-version }}"
+          exit 1
+  macOS-direct-install:
+    name: Install Pulumi via script on macOS
+    runs-on: macos-latest
+    steps:
+      - run: curl -fsSL https://get.pulumi.com | sh
+      - run: echo "/Users/runner/.pulumi/bin" >> $GITHUB_PATH
+      - name: Pulumi Version Details
+        id: vars
+        run: |
+          echo ::set-output name=installed-version::$(pulumi version)
+          echo ::set-output name=expected-version::v$(curl -sS https://www.pulumi.com/latest-version)
+      - run: which pulumi
+      - name: Error if incorrect version found
+        if: steps.vars.outputs.expected-version != steps.vars.outputs.installed-version
+        run: |
+          echo "Expected version ${{ steps.vars.outputs.expected-version }} but found ${{ steps.vars.outputs.installed-version }}"
+          exit 1
+  macos-verify-download-link:
+    name: Verify Direct Download link on macOS
+    runs-on: macos-latest
+    steps:
+      - name: Direct Download
+        run: curl -L -o pulumi.tar.gz https://get.pulumi.com/releases/sdk/pulumi-v$(curl -sS https://www.pulumi.com/latest-version)-darwin-x64.tar.gz
+      - run: ls -la
+  linux-direct-install:
+    name: Install Pulumi via script on Ubuntu
+    runs-on: ubuntu-latest
+    steps:
+      - name: Remove existing version
+        run: sudo rm /usr/local/bin/pulumi
```

Comment:
Reading the documentation, `echo ... >> $GITHUB_PATH` prepends, in which case, won't the downloaded pulumi supersede this and we don't need to remove?

Reply:
I am trying to ensure there isn't anything else clouding the test :) This is a first iteration of it that I will ensure is cleaner when I know it's working well